apps/back-end/src/app/auth/auth.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { AmrsAuthService } from './amrs-auth.service';
+import { AmrsAuthResponse, JwtAmrsUser } from './dto/amrs-auth.dto';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private amrsAuthService: AmrsAuthService,
+    private jwtService: JwtService
+  ) {}
+
+  async validateUser(
+    username: string,
+    password: string
+  ): Promise<AmrsAuthResponse> {
+    const amrsUser = await this.amrsAuthService.authenticate({
+      username,
+      password,
+    });
+    if (!amrsUser) throw new NotFoundException('Wrong username or password!');
+    if (!amrsUser.authenticated) {
+      throw new NotFoundException('Wrong username or password!');
```

Comment:
This can be shortened.
`if (!amrsUser || !amrsUser.authenticated) throw new NotFoundException('Wrong username or password!');`